Sources/StructuredQueriesMacros/Internal/PatternBindingSyntax.swift

```diff
@@ -37,4 +37,13 @@ extension PatternBindingSyntax {
     optionalized.typeAnnotation?.type = optionalType
     return optionalized
   }
+
+  func isOptional() -> Bool {
+    // x: Optional<T> or x: T?
+    if self.typeAnnotation?.type.isOptionalType == true { return true }
+    // Missing cases
+    // x = Optional<T>.some(_)
+    // x = fnReturningOptionalType()
```

Comment:
Is this helper mostly here to mark these cases as todos?

Reply:
I forgot to mention in the PR description, but I was wondering if it's worth it trying to do anything to figure out if a type is actually Optional or not. I was getting in a rabbit hole and left it as is to not waste too much time and get some input.

It doesn't seem worth it IMO, but I left there to inform that `isOptional` does not guarantee that the binding is actually optional, and left some examples of bindings that could return a false negative.